### ISSUE TITLE
feat: look for `try`s on falliable functions with empty error unions

### ISSUE THREAD

**Issue by DonIsaac** [DonIsaac/zlint#153]
This pathological code compiles when it shouldn't.

```zig
const std = @import("std");

fn foo() !u32 {
 return 1;
}

fn bar() void {
  const i = try foo();
  std.debug.print("foo() returns {d}\n", .{i});
}
```

We should add a rule checking falliable functions and make sure they do one of the following:
1. Return an error
2. Use `try`
3. Return a call to a function that returns an error union

Item 3 is not fully doable without a type checker, but we can allow cases that return any function call.

**Comment by DonIsaac** [comment]
this has been implemented
